BasicClasses/BasicClasses.tex

```diff
@@ -881,13 +896,59 @@ \section{Exceptions}
 		ensure: [Transcript showln: 'My execution is ensured.'].
 \end{code}
 
-% - Exception sets
-% - Resuming
-% - Passing along state
-% As we sometimes want to catch several kinds of exceptions, you can combine \ct{Exception}s into an \ct{ExceptionSet} by sending the message \ct{Exception>>>,}.
+Note that under normal circumstraints, you should never place a return statement (\ct{^}) in an ensure block.
+If you do so, the execution will not reach further \ct{ensure:} blocks but will continue as if no exception had occurred.
+In particular, this can create a zombie process when you abandon a program from a debugger.
+
+So far, we only have \emph{returned} from an exception.
+However, exception handlers have a couple of other options for dealing with exceptions by sending messages to them:
+
+\begin{itemize}
+    \item \mthind{Exception}{return:} returns a value from the handler block.
```

Comment:
"returns a value from the handler block." this might be somewhat misleading, as it sounds like the handler block retuns something, but you mean that the value stems from the handler block and is returned at the signaling location, right?

Reply:
oh, I confused #return: with #resume:. Why does #return: exist? Is there any use case that is not meta? The example below is confusing (at least to me) in that regard. I think because the execption is some form of continuation for me, so if I send a message to it, I refer to that line of execution (Might also be the result of me knowing that #resume: exists).

Maybe move #return: to the bottom and mention that "you can also explicitly return a value from the handler"?

Reply:
Good question. Here are some ideas regarding the relevance of `return:`

- First, `return:` is used in `Context>>#handleSignal:`.
- Second, the reason why this is used in many regular handler blocks might be historic (???).
- Third, `return:` can be a handy convenience for writing early exists in larger handler blocks.
- Fourth, you could theoretically pass the exception to another object and have it return from there.

Still, "returns a value from the handler block" is incorrect, it should rather be "returns a value from the handler context"/"from the exception handler". Yes, we could also move this down, but because the description contains a backreference to the implicit return, I thought it would be less confusing at the top of the list? What do you think? :)

Reply:
Keep it, but to the end of the list.